transcom/mymove
# MB-16221: add middleware to internal api to ensure users are service members

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16221)

## Summary

This middleware has an allow list for endpoints that can't enable the stricter check as of this moment, with the goal of getting rid of the list eventually. This closes a security loophole, but it should not negatively impact regular user experience or usage.

It's a bit difficult to check the failure state via the site because we automatically create service member records for new users. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Run the app as a customer and ensure you can still navigate the customer app successfully.
2. Run tests to ensure there aren't any failures.

### Backend
